staging/src/k8s.io/api/coordination/v1alpha1/types.go

```diff
@@ -0,0 +1,100 @@
+	// +optional
+	Spec LeaseCandidateSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// LeaseSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// LeaseName is the name of the lease for which this candidate is contending.
+	// This field is immutable.
+	// +required
+	LeaseName string `json:"leaseName" protobuf:"bytes,1,name=leaseName"`
+	// CandidateDurationSeconds is the duration that candidates are "valid"
+	// for. If it is within this interval from the renewTime, leader election will not ask this LeaseCandidate to renew.
+	// +required
+	CandidateDurationSeconds int32 `json:"candidateDurationSeconds" protobuf:"varint,2,opt,name=candidateDurationSeconds"`
+	// PingTime is the last time that the server has requested the LeaseCandidate
+	// to renew. It is only done during leader election to check if any
+	// LeaseCandidates have become ineligible. When PingTime is updated, the
+	// LeaseCandidate will respond by updating RenewTime.
+	// +optional
+	PingTime *metav1.MicroTime `json:"pingTime,omitempty" protobuf:"bytes,3,opt,name=pingTime"`
+	// RenewTime is the time that the LeaseCandidate was last updated.
+	// Unlike Lease objects, candidates are not refreshed frequently.
+	// Any time a Lease needs to do leader election, the Ping field
+	// is updated to signal to the LeaseCandidate that they should update
+	// the RenewTime.
+	// +optional
+	RenewTime *metav1.MicroTime `json:"renewTime,omitempty" protobuf:"bytes,4,opt,name=renewTime"`
+	// BinaryVersion is the binary version. It must be in a semver format.
+	// This field is effectively required when Strategy is "OldestEmulationVersion"
+	// +optional
+	BinaryVersion string `json:"binaryVersion,omitempty" protobuf:"bytes,5,opt,name=binaryVersion"`
+	// EmulationVersion is the emulation version. It must be in a semver format.
+	// This field is effectively required when Strategy is "OldestEmulationVersion"
+	// +optional
+	EmulationVersion string `json:"emulationVersion,omitempty" protobuf:"bytes,6,opt,name=emulationVersion"`
+	// PreferredStrategies indicates the list of strategies for picking the leader for coordinated leader election.
+	// The list is ordered, and the first strategy supersedes all other strategies. The list is used by coordinated
+	// leader election to make a decision about the final election strategy. This follows as
+	// - If all clients have strategy X as the first element in this list, strategy X will be used.
+	// - If a candidate has strategy [X] and another candidate has strategy [Y, X], Y supersedes X and strategy Y
+	//   will be used
+	// If
+	// (Alpha) Using this field requires the CoordinatedLeaderElection feature gate to be enabled.
+	// +featureGate=CoordinatedLeaderElection
+	// +listType=atomic
+	// +required
+	PreferredStrategies []v1.CoordinatedLeaseStrategy `json:"preferredStrategies,omitempty" protobuf:"bytes,7,opt,name=preferredStrategies"`
```

Comment:
When CLE controiller sees  a pool of candidates and the referenced Lease does not exist, itr(CLE) will create the lease, right?  In creating the lease it needs to know:
- name (from `LeaseName` here)
- strategy (from `PreferredStrategies` here)
- duration (from ???)

How does CLE know what to set the Lease's duration to?  Is it hard-coded?  Or should candidate specify their preferred duration, and CLE controller decides which one to use?  How does it decide?

Reply:
Okay that was the original use case for spec.LeaseDurationSeconds in the candidate and that somehow got forgotten through our iterations. I think it makes sense to keep that field as the preferred duration for the lease that CLE can use.